app_clients/pod-delete/pod-delete.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/sh
+
+set -e
+
+## mandatory arg
+container_id=${CONTAINER_ID}
+interval=${INTERVAL:=5}
+force=${FORCE:=false}
+app_ns=${APP_NS}
+kill_count=${KILL_COUNT}
+app_label=${APP_LABEL}
+iterations=${ITERATIONS}
+
+k=0
+until [ "$k" -ge "${iterations}" ]
+do
+k=$((k+1))
+## When the kill count is not defined choose any single random pod with the given label and namesapce
+if [[ -z ${kill_count} ]]; then
+echo "[Inject]: Kill a random application"
+rand_pod=$(kubectl get pod -n ${app_ns} -l ${app_label} -o=custom-columns=NAME:".metadata.name" --no-headers | shuf -n1)
+app_pod=${rand_pod}
+else
+# When kill count is defined select the equal number of pod for chaos with given namespace and label
+echo "[Inject]: Starting experiment with kill count value: ${kill_count}"
+pod_list=$(kubectl get pod -n ${app_ns} -l ${app_label} -o=custom-columns=NAME:".metadata.name" --no-headers | shuf -n${kill_count})
+app_pod=${pod_list}
+fi
+
+## printing the name of application pod to be killed
+echo "Name of application pod to be killed: ${app_pod}"
+
+###########################################################
+###############    FORCE POD DELETE      ##################
+###########################################################
+
+## killing the application pod forcefully if force is set to true
+if [[ "${force}" == "true" ]]
+then
+echo "[Inject]: Killing the application pod forcefullly"
+kubectl delete pod -n ${app_ns} --force --grace-period=0 --wait=false ${app_pod}
+
+echo "[Status]: Verification for the recreation of application pod"
+## checking the status of pod and wait for it to come in running state
+echo "[Status]: Checking the status of pods"
+n=0
+until [ "$n" -ge 90 ]
+do
+pod_status=$(kubectl get pods -n ${app_ns} -l ${app_label} -o custom-columns=:.status.phase --no-headers | uniq)
+    if [[ ${pod_status} == "Running" ]]
+    then
+        break;
+    else
+        n=$((n+1)) 
+        echo "pod is in ${pod_status} state please wait"
+        sleep 2
+    fi
+done
```

Comment:
if after 90 iterations, the pod is still not in running state then it will fail or continue its execution?

Reply:
yes it will fail because of `set -e` used ...